### ISSUE TITLE
Biometrics service crash

### ISSUE THREAD

**Issue by ItsTimeTraveler** [PixelExperience/android-issues#682]
## Build date

```
PixelExperience_sailfish-11.0-20210913-1612-OFFICIAL
```

## Expected Behavior

```
Fingerprint scanner working, no crashes of system services.
```

## Current Behavior

```
I see crashes every 2-3 seconds.

09-14 00:30:29.050  8608  8608 I crash_dump64: obtaining output fd from tombstoned, type: kDebuggerdTombstone
09-14 00:30:29.051   657   657 I tombstoned: received crash request for pid 8604
09-14 00:30:29.052  8608  8608 I crash_dump64: performing dump of process 8604 (target tid = 8604)
09-14 00:30:29.055  8608  8608 F DEBUG   : *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
09-14 00:30:29.055  8608  8608 F DEBUG   : Build fingerprint: 'google/sailfish/sailfish:10/QP1A.191005.007.A3/5972272:user/release-keys'
09-14 00:30:29.055  8608  8608 F DEBUG   : Revision: '0'
09-14 00:30:29.055  8608  8608 F DEBUG   : ABI: 'arm64'
09-14 00:30:29.056  8608  8608 F DEBUG   : Timestamp: 2021-09-14 00:30:29-0400
09-14 00:30:29.056  8608  8608 F DEBUG   : pid: 8604, tid: 8604, name: android.hardwar  >>> /vendor/bin/hw/android.hardware.biometrics.fingerprint@2.1-service <<<
09-14 00:30:29.057  8608  8608 F DEBUG   : uid: 1000
09-14 00:30:29.057  8608  8608 F DEBUG   : signal 6 (SIGABRT), code -1 (SI_QUEUE), fault addr --------
09-14 00:30:29.057  8608  8608 F DEBUG   : Abort message: 'Attempted to retrieve value from failed HIDL call: Status(EX_TRANSACTION_FAILED): '-9 (Bad file descriptor): ''
09-14 00:30:29.057  8608  8608 F DEBUG   :     x0  0000000000000000  x1  000000000000219c  x2  0000000000000006  x3  0000007fe63e8710
09-14 00:30:29.057  8608  8608 F DEBUG   :     x4  0000000000000000  x5  0000000000000000  x6  0000000000000000  x7  7f7f7f7f7f7f7f7f
09-14 00:30:29.057  8608  8608 F DEBUG   :     x8  00000000000000f0  x9  431e7263d7458cda  x10 ffffff80ffffffdf  x11 ffffffc0ffffffdf
09-14 00:30:29.057  8608  8608 F DEBUG   :     x12 0000007fe63e86b0  x13 000000000000006f  x14 00000001c47dd73e  x15 00010e25e6c24202
09-14 00:30:29.057  8608  8608 F DEBUG   :     x16 00000070b5791c78  x17 00000070b5773c20  x18 00000070b65c8000  x19 00000000000000ac
09-14 00:30:29.057  8608  8608 F DEBUG   :     x20 000000000000219c  x21 00000000000000b2  x22 000000000000219c  x23 00000000ffffffff
09-14 00:30:29.057  8608  8608 F DEBUG   :     x24 0000000000000000  x25 0000000000000000  x26 0000000000000000  x27 0000000000000000
09-14 00:30:29.057  8608  8608 F DEBUG   :     x28 0000000000000000  x29 0000007fe63e8790
09-14 00:30:29.058  8608  8608 F DEBUG   :     lr  00000070b572556c  sp  0000007fe63e86f0  pc  00000070b5725598  pst 0000000000000000
09-14 00:30:29.077  8608  8608 F DEBUG   : backtrace:
09-14 00:30:29.077  8608  8608 F DEBUG   :       #00 pc 000000000004e598  /apex/com.android.runtime/lib64/bionic/libc.so (abort+176) (BuildId: 6fef2b0ca43a1525af04ae1c6da427de)
09-14 00:30:29.077  8608  8608 F DEBUG   :       #01 pc 00000000000062a0  /system/lib64/liblog.so (__android_log_default_aborter+12) (BuildId: b5e84530b2278e3780a02ed82997eafa)
09-14 00:30:29.078  8608  8608 F DEBUG   :       #02 pc 0000000000013058  /system/lib64/libbase.so (android::base::LogMessage::~LogMessage()+316) (BuildId: 49b27a71402b491c2510d16fc20009fe)
09-14 00:30:29.078  8608  8608 F DEBUG   :       #03 pc 00000000000542e8  /system/lib64/libhidlbase.so (android::hardware::details::return_status::onValueRetrieval() const+236) (BuildId: 45e13396d96b7f3ef5a7ead15087e97b)
09-14 00:30:29.078  8608  8608 F DEBUG   :       #04 pc 000000000005d288  /system/lib64/libhidlbase.so (android::hardware::details::registerAsServiceInternal(android::sp<android::hidl::base::V1_0::IBase> const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)+184) (BuildId: 45e13396d96b7f3ef5a7ead15087e97b)
09-14 00:30:29.078  8608  8608 F DEBUG   :       #05 pc 0000000000019b00  /system/lib64/android.hardware.biometrics.fingerprint@2.1.so (android::hardware::biometrics::fingerprint::V2_1::IBiometricsFingerprint::registerAsService(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)+92) (BuildId: 91fc3b69c01578acf2ec095ab3f80290)
09-14 00:30:29.078  8608  8608 F DEBUG   :       #06 pc 0000000000003e68  /vendor/bin/hw/android.hardware.biometrics.fingerprint@2.1-service (main+144) (BuildId: 08c1183d75a10f457089ab47133c642c)
09-14 00:30:29.078  8608  8608 F DEBUG   :       #07 pc 0000000000049aec  /apex/com.android.runtime/lib64/bionic/libc.so (__libc_init+108) (BuildId: 6fef2b0ca43a1525af04ae1c6da427de)
09-14 00:30:29.111  1139  1579 W NativeCrashListener: Couldn't find ProcessRecord for pid 8604
09-14 00:30:29.112  8608  8608 E crash_dump64: AM data write failed: Broken pipe
09-14 00:30:29.113   657   657 E tombstoned: Tombstone written to: /data/tombstones/tombstone_11
```

## Possible Solution

```
No solution found.
```

## Commentary

First off, thank you for maintaining the OG Pixel. I did not check on the earlier builds, but it appears that fingerprint service now crashes every 2-3 seconds. I should note that I have dropped my phone just before doing the upgrade, so it might be a HW issue. The phone survived and still in good shape (no cracks or whatsoever), but fingerprint scanner does not work, there's even no option to add it in the settings.

/device sailfish
/version eleven

**Comment by pixelexperiencebot** [comment]
Validation failed, this issue will be closed for not complying with the established standards.

Errors:
- Invalid issue template

**Comment by ItsTimeTraveler** [comment]
(moved from the OP)

First off, thank you for maintaining the OG Pixel. I did not check on the earlier builds, but it appears that fingerprint service now crashes every 2-3 seconds. I should note that I have dropped my phone just before doing the upgrade, so it might be a HW issue. The phone survived and still in good shape (no cracks or whatsoever), but fingerprint scanner does not work, there's even no option to add it in the settings.
